app/javascript/controllers/complex_checkbox_controller.js

```diff
@@ -0,0 +1,46 @@
+import { Controller } from "stimulus"
+
+export default class extends Controller {
+  static targets = ["checkbox", "checkedSection", "uncheckedSection"]
+
+  connect() {
+    this.toggleInputs()
+  }
+
+  toggleInputs(event) {
+    if(this.checkboxTarget.checked) {
+      this.toggle(this.checkedSectionTarget, this.uncheckedSectionTarget, !!event)
+    } else {
+      this.toggle(this.uncheckedSectionTarget, this.checkedSectionTarget, !!event)
+    }
+  }
+
+  toggle(target, source, copy) {
+    target.hidden = false
+    source.hidden = true
+
+    const targetInputs = target.querySelectorAll('select,input[type="number"],input[type="checkbox"]')
+    targetInputs.forEach((element) => element.disabled = false)
+
+    const sourceInputs = source.querySelectorAll('select,input[type="number"],input[type="checkbox"]')
+    sourceInputs.forEach((element) => element.disabled = true)
+
+    if(!copy) return
+
+    // Copy source values to target
+    for (let i = 0; i < Math.min(targetInputs.length, sourceInputs.length); i++) {
+      if(targetInputs[i].type == 'checkbox') {
+        targetInputs[i].checked = sourceInputs[i].checked
+      } else {
+        targetInputs[i].value = sourceInputs[i].value
+      }
+
+      // This triggers the other controllers.
```

Comment:
which is needed for what purpose?

Reply:
To perform validation, etc. I updated the comment.